### PR TITLE
ci(gha/release): download artifacts from annotated tags

### PR DESCRIPTION
**Pull request by tiagolobocastro**
Annotated tags need an extra deref to fetch the commit sha.

Ensures the publish of release artifacts works with an annotated tag.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.

### CODE REVIEW

**Review by pchandra19 — APPROVED** [review]

**Review by sinhaashish — APPROVED** [review]

File: scripts/github/release-artifacts.sh [hunk 1]
```diff
@@ -34,8 +34,18 @@ get_artifacts_id() {
...
-  job=$(gh run list --branch "$tag" --workflow "$workflow" --repo "$repo" --json conclusion,headSha,databaseId --jq ".[] | select(.headSha == $sha)")
+  local tag_ref=$(gh api repos/$repo/git/refs/tags/$tag)
+  local type=$(echo $tag_ref | jq -r '.object.type')
+  if [ "$type" = "tag" ]; then
+    # Git annotated tags are an object on their own, so we need
+    # another step to fetch the commit sha.
+    local sha=$(echo $tag_ref | jq -r '.object.sha')
+    tag_ref=$(gh api repos/$repo/git/tags/$sha)
+  elif [ "$type" != "commit" ]; then
+    die "Invalid tag type: $type"
```
Comment (niladrih) [inline scripts/github/release-artifacts.sh:45]: Helper functions, if you're interested: https://github.com/openebs/mayastor-extensions/blob/develop/scripts/utils/log.sh

File: scripts/github/release-artifacts.sh [hunk 2]
```diff
@@ -34,8 +34,18 @@ get_artifacts_id() {
...
+  fi
+  local sha=$(echo $tag_ref | jq '.object.sha')
```
Comment (niladrih) [inline scripts/github/release-artifacts.sh:47]:
```suggestion
  local sha=$(echo $tag_ref | jq -r '.object.sha')
```

**Review by niladrih — APPROVED** [review]

File: scripts/github/release-artifacts.sh [hunk 3]
```diff
@@ -161,7 +172,6 @@ while [ "$#" -gt 0 ]; do
     upload)
       shift
       COMMAND="upload"
-      shift
```
Comment (niladrih) [inline scripts/github/release-artifacts.sh:164]: Why is this shift not needed?

**Review by niladrih — COMMENTED** [review]
Left one comment

File: scripts/github/release-artifacts.sh (hunk 3 above)
Comment (tiagolobocastro) [reply scripts/github/release-artifacts.sh:164]: Because it's not taking any arguments

File: scripts/github/release-artifacts.sh (hunk 2 above)
Comment (tiagolobocastro) [reply scripts/github/release-artifacts.sh:47]: No, we need the quotes.

File: scripts/github/release-artifacts.sh (hunk 3 above)
Comment (tiagolobocastro) [reply scripts/github/release-artifacts.sh:164]: I removed the other shift ti make it clearer

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**